indexer-app/src/cmr/indexer/services/index_service.clj

```
@@ -697,50 +697,70 @@
   (fn [_context concept-id _revision-id _options]
     (cs/concept-id->type concept-id)))
 
-(defmethod delete-concept :default
-  [context concept-id revision-id options]
-  ;; Assuming ingest will pass enough info for deletion
-  ;; We should avoid making calls to metadata db to get the necessary info if possible
+(defn- delete-concept-default-helper
+  [context concept concept-id revision-id options]
+  (if (nil? concept)
+    (errors/throw-service-error
```

will this code ever run given that (when) is used below?

---

The "delete-concept :default" func can be called when other concepts like granules, collections, etc are deleted and if they aren't found in the db we'll want to throw this error here